Doc/c-api/refcounting.rst

```diff
@@ -119,3 +119,32 @@ The following functions or macros are only for use within the interpreter core:
 :c:func:`_Py_Dealloc`, :c:func:`_Py_ForgetReference`, :c:func:`_Py_NewReference`,
 as well as the global variable :c:data:`_Py_RefTotal`.
 
+.. _immortal-objects:
+
+Immortal Objects
+================
+
+"Immortal" objects are those that are expected to never be deallocated
+by the runtime (due to their reference count reaching 0).  In the public
+C-API examples of such objects includes the singletons and many builtin
+types.  For such objects the reference count is essentially irrelevant.
+Immortal objects are especially useful if otherwise immutable.
+
+Note that, for now, the API for immortal objects is not available
+for general use and it should not be considered stable yet.
+
+.. c:function:: int _PyObject_IsImmortal(PyObject *o)
+
+   Return non-zero if the object is immortal.
+
+   .. versionadded:: 3.10
+
+.. c:function:: void _PyObject_SetImmortal(PyObject *o)
+
+   Mark an object as immortal.
+
+   .. versionadded:: 3.10
+
+Also see :c:macro:`_PyObject_HEAD_IMMORTAL_INIT` and
+:c:macro:`_PyVarObject_HEAD_IMMORTAL_INIT`.
```

Comment:
I don't think that we document API which are parts of the internal C API. If you want to document it, mention that it's only part of the internal C API.

Reply:
We do document some internal API, for the sake of core development.  However, the docs I added don't add much value in that regard.  I was thinking more of them as eventually public, which I pulled out of this PR, so I'll remove the docs.